app/models/measurement_list_reader.rb

```diff
@@ -18,74 +18,39 @@ def load
 
     @measurements = Measurement.where(mcc_filter: [nil, mcc])
     @measurements = filter_by_show_hide
-    # it might be nice to run this each separate threads since they do 3 GR calls each
-    # Spencer suggests waiting until we test how the performs
-    # https://github.com/tra/spawnling might be a good option, or a vanilla thread pool
-    # https://blog.engineyard.com/2014/ruby-thread-pool
-    @measurements.each(&method(:load_gr_value))
+    load_measurements_from_gr
     split_children
   end
 
   private
 
-  def load_gr_value(measurement)
-    return load_historic_gr_values(measurement) if historical
-    levels = Power.current.try(:measurement_levels) || [:total, :local, :person]
-    levels.each do |level|
-      measurement_level_id = measurement.send("#{level}_id")
-      filter_params = gr_request_params(level)
-      gr_resp = GlobalRegistryClient.client(:measurement_type).find(measurement_level_id, filter_params)
-      value = gr_resp['measurement_type']['measurements'].map { |m| m['value'].to_f }.sum
-      measurement.send("#{level}=", value)
+  def load_measurements_from_gr
+    work_q = Queue.new
+    params = gr_loading_params
+    @measurements.each { |m| work_q.push m }
+    workers = (1..ENV['MEASUREMENT_THREAD_COUNT'].to_i).map do
+      Thread.new do
+        until work_q.empty?
+          measurement = work_q.pop(true)
+          Measurement::GrValueLoader.new(params.merge(measurement: measurement)).load_gr_value
+        end
+      end
     end
+    workers.map(&:join)
```

Comment:
Are you using the output of the calls to `join`?  If not, could you use `each` instead (no allocated list for the output and feels slightly clearer to me to show you aren't using the output)?